packages/core/src/test/awsService/appBuilder/wizards/deployTypeWizard.test.ts

```diff
@@ -0,0 +1,121 @@
+        const choices = await deployTypeWizard.run()
+        // Then
+        assert(!choices)
+    })
+
+    it('deploy is selected', async function () {
+        /**
+         * This test focus on that deploy wizard get triggered when customer choose to use sam deploy
+         * Selection for deploy wizard speficy here focus on only one case
+         * More cases are test in Deploy.test.ts
+         *
+         */
+        PrompterTester.init()
+            .handleQuickPick('Select deployment command', async (picker) => {
+                await picker.untilReady()
+                assert.strictEqual(picker.items[0].label, 'Sync')
+                assert.strictEqual(picker.items[1].label, 'Deploy')
+                assert.strictEqual(picker.items.length, 2)
+                picker.acceptItem(picker.items[1])
+            })
+            .handleInputBox('Specify SAM parameter value for SourceBucketName', (inputBox) => {
+                inputBox.acceptValue('my-source-bucket-name')
+            })
+            .handleInputBox('Specify SAM parameter value for DestinationBucketName', (inputBox) => {
+                inputBox.acceptValue('my-destination-bucket-name')
+            })
+            .handleQuickPick('Select a SAM/CloudFormation Template', async (quickPick) => {
+                // Need sometime to wait for the template to search for template file
+                await quickPick.untilReady()
+                quickPick.acceptItem(quickPick.items[0])
+            })
+            .handleQuickPick('Specify parameters for deploy', async (quickPick) => {
+                // Need time to check samconfig.toml file and generate options
+                await quickPick.untilReady()
+                assert.strictEqual(quickPick.items[2].label, 'Use default values from samconfig')
+                quickPick.acceptItem(quickPick.items[2])
+            })
+            .build()
+        deployTypeWizard = new DeployTypeWizard(syncWizard, deployWizard)
+        const choices = await deployTypeWizard.run()
+        // Then
+        assert.strictEqual(choices?.choice, 'deploy')
+    })
+
+    it('sync is selected', async function () {
+        /**
+         * This test focus on that sync wizard get triggered when customer choose to use sam sync
+         * Selection for deploy wizard speficy here focus on only one case
+         * More cases are test in Sync.test.ts
+         *
+         */
+        PrompterTester.init()
+            .handleQuickPick('Select deployment command', async (picker) => {
+                await picker.untilReady()
+                assert.strictEqual(picker.items[0].label, 'Sync')
+                assert.strictEqual(picker.items[1].label, 'Deploy')
+                assert.strictEqual(picker.items.length, 2)
+                picker.acceptItem(picker.items[0])
+            })
+            .handleQuickPick('Select a SAM/CloudFormation Template', async (quickPick) => {
```

Comment:
There's a potential issue here, if `Select a SAM/CloudFormation Template` is not shown at all, the current test logic seems will pass without any error

Reply:
Understood. I can implement additional validation in PrompterTester class to cover this missing prompter case.

Reply:
Added additional validation and update the tests.